vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,376 @@
+                ${this.getIndexUrl(this.versionResolver.getMajorMinor(version))}.
+                The json does not have the parameter ${this.releasesSdkNameKey} which means the API publisher has published invalid dotnet release data. Please file an issue at https://github.com/dotnet/vscode-dotnet-runtime.`));
+            this.eventStream.post(err);
+            throw err.error;
+        }
+
+        let desiredFileExtension = '';
+
+        switch(operatingSystemInDotnetFormat)
+        {
+            case 'win': {
+                desiredFileExtension = '.exe';
+                break;
+            }
+            case 'osx': {
+                desiredFileExtension = '.pkg';
+                break;
+            }
+            case 'linux': {
+                desiredFileExtension = '.gz';
+                break;
+            }
+            default:
+            {
+                const err = new DotnetUnexpectedInstallerOSError(new Error(`The SDK Extension failed to map the OS ${operatingSystemInDotnetFormat} to a proper package type.
+                    Your architecture: ${os.arch()}. Your OS: ${os.platform()}.`));
+                this.eventStream.post(err);
+                throw err.error;
+            }
+        }
+
+        return path.extname(installerFileName) === desiredFileExtension;
+    }
+
+    /**
+     *
+     * @param version the non-specific version, such as 6.0.4xx.
+     * @param band The band of the version.
+     */
+    private async getNewestSpecificVersionFromFeatureBand(version : string) : Promise<string>
+    {
+        const band : string = this.versionResolver.getFeatureBandFromVersion(version);
+        const indexUrl : string = this.getIndexUrl(this.versionResolver.getMajorMinor(version));
+
+        // Get the sdks
+        const indexJson =  await this.fetchJsonObjectFromUrl(indexUrl);
+        const releases = indexJson[this.releasesJsonKey]
+
+        if(releases.length === 0)
+        {
+            const badJsonErr = new DotnetInvalidReleasesJSONError(new Error(`${this.releasesJsonErrorString}${indexUrl}`));
+            this.eventStream.post(badJsonErr);
+            throw badJsonErr.error;
+        }
+
+        // Assumption: The first release in releases will be the newest release and contain the newest sdk for each feature band. This has been 'confirmed' with the releases team.
+        const sdks = releases[0][this.releasesSdksKey];
+        for (const sdk of sdks)
+        {
+            // The SDKs in the index should be in-order, so we can rely on that property.
+            // The first one we find with the given feature band will also be the 'newest.'
+            const thisSDKVersion : string = sdk[this.releasesSdkVersionKey];
+            if(this.versionResolver.getFeatureBandFromVersion(thisSDKVersion) === band)
+            {
+                return thisSDKVersion;
+            }
+        }
+
+        // TODO: make a test for this error msg
```

Comment:
Has this TODO been completed?

Reply:
It has not, I did manually test and it seems OK. I think I will create an issue to make this so we can merge this. https://github.com/dotnet/vscode-dotnet-runtime/issues/1442 Good catch on me almost commiting an evil thing which is merging a TODO 😆